src/ClapCommandLineHandler.package/ClapSubclassHandlerCollectorStrategy.class/instance/collectHandlers.st

```diff
@@ -0,0 +1,7 @@
+as yet unclassified
+collectHandlers
+	"I look in the subclasses of ClapCommandLineHandler"
+	| allCommands |
+	allCommands := ClapCommandLineHandler allSubclasses select: [ :subclass | subclass hasCommand].
+	allCommands ifEmpty: [^ nil].
```

Comment:
I missed that… this is _really_ **really** ugly… 😷

Returning an empty collection should be perfectly fine! Instead, by adding a special-case here, you're forcing the caller to add a check, because now the return value is either a collection or `nil`, which are not polymorphic in any useful way…

In many cases you'd make special efforts to avoid `nil` and implement special objects that represent absent or incorrect values but are still polymorphic enough with expected values. This the _Null Object_ design pattern, and what I started doing in Clap with `ClapMismatch`.